packages/core/contracts-ovm/insured-bridge/implementation/BridgePoolFactory.sol

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+// Importing local copies of OVM contracts is a temporary fix until the @eth-optimism/contracts package exports 0.8.x
+// contracts. These contracts are relatively small and should have no problems porting from 0.7.x to 0.8.x, and
+// changing their version is preferable to changing this contract to 0.7.x and defining compatible interfaces for all
+// of the imported DVM contracts below.
+import "./OVM_CrossDomainEnabled.sol";
+import "./BridgePoolFactoryInterface.sol";
+import "../../../contracts/oracle/interfaces/IdentifierWhitelistInterface.sol";
+import "../../../contracts/oracle/interfaces/FinderInterface.sol";
+import "../../../contracts/oracle/implementation/Constants.sol";
+import "../../../contracts/common/interfaces/AddressWhitelistInterface.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @notice Administrative contract deployed on L1 that has an implicit reference to a DepositBox. This contract is
+ * responsible for deploying new BridgePools, which houses passive liquidity and enables relaying of L2 deposits.
+ * @dev The owner of this contract can call permissioned functions on the L2 DepositBox.
+ */
+contract BridgePoolFactory is BridgePoolFactoryInterface, Ownable, OVM_CrossDomainEnabled {
+    // Finder used to point to latest OptimisticOracle and other DVM contracts.
+    address private finder;
+
+    // L2 Deposit contract that originates deposits that can be fulfilled by this contract.
+    address private depositContract;
+
+    mapping(address => L1TokenRelationships) private whitelistedTokens;
+
+    // Set upon construction and can be reset by Owner.
+    uint256 private optimisticOracleLiveness;
+    uint256 private proposerBondPct;
+    bytes32 private identifier;
```

Comment:
why make these private and then make the accessors? why not just make them public?

Reply:
Because the children can't access these methods if they are public, so rather than have redundant state variables and getter methods for the children to read, I chose to just use getters. Does that make sense?

Reply:
why can the children not just call the default public accessors associated with public state variables?

Reply:
I ideally wanted to just have to import the interface of `BridgePoolFactory` into `BridgePool`, but I removed the public getters for now to reduce code clutter.

Reply:
@nicholaspai you can declare public getters in the interface and override it using state variables:

> Public state variables can override external functions if the parameter and return types of the function matches the getter function of the variable:
```
// SPDX-License-Identifier: GPL-3.0
pragma solidity >=0.6.0 <0.9.0;

contract A
{
    function f() external view virtual returns(uint) { return 5; }
}

contract B is A
{
    uint public override f;
}
```
Source: https://docs.soliditylang.org/en/v0.8.6/contracts.html#function-overriding